code/modules/mapping/mapping_helpers.dm

```diff
@@ -1359,3 +1359,35 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
 
 	engraved_wall.AddComponent(/datum/component/engraved, engraving["story"], FALSE, engraving["story_value"])
 	qdel(src)
+
+/obj/effect/mapping_helpers/emergency_shuttle_event_override
+	name = "emergency shuttle events override"
+	icon_state = "shuttleeventoverride"
+	late = TRUE
+	/// List of event paths to run
+	var/list/events_override = list()
```

Comment:
This shouldn't be it's own object. You can put all this logic on the shuttle template itself. There's actually a proc on the template called `post_load(obj/docking_port/mobile/M)` which is perfect for your application, you don't even need to find the port

You should put this code on the `/datum/map_template/shuttle/emergency` template (you can also put it on /shuttle, but shuttle events aren't properly set up for that yet)